### ISSUE TITLE
No IPathImportMapper adapter for imp.NullImporter

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1973]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1973|
|Type|defect|
|Created|2006-07-30 20:31:03Z|

This totally breaks trial in Python 2.5 trunk@HEAD

<details><summary>Searchable metadata</summary>

```
trac-id__1973 1973
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1154291463000000 1154291463000000
changetime__1155298819000000 1155298819000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

(In [17769]) deal with NullImporter if we have to

refs [#1973](https://github.com/twisted/twisted/issues/1973)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

Is there a final word on what Python is actually going to look like in the 2.5 release?  I haven't been keeping up with python-dev but it looks like there was still some discussion.

**Comment by twisted-trac** [comment]
|<img alt="scmikes's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|scmikes commented|
|-|-|

Glyph,

I just installed python beta3 (Released on Aug 3), on the suse buildbot, and this behaviour is still in place.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @radix|
|-|-|

python-dev says the current behavior is final.  radix will review this for me.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_5"></a>|@radix set owner to @exarkun|
|-|-|

exarkun said this needs more work.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @radix|
|-|-|

Okay, really ready for review now.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_7"></a>|@radix set owner to @exarkun|
|-|-|

There aren't any unit tests

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_8"></a>|@radix commented|
|-|-|

Ok, the test you added is satisfying. Go ahead and merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|


Merge ignore-warnings-1972

Author: exarkun
Reviewer: radix
Fixes [#1973](https://github.com/twisted/twisted/issues/1973)

This changes the custom importer support in twisted.python.modules to deal
with any unknown importer by passing the path straight through.  This will
not allow arbitrary importers to work correctly with twisted.python.modules,
but it will prevent them from completely disrupting uses of it.  Support for
other importers should be added when they see more widespread use.  Third
parties can already support custom importers by registering adapters for
IPathImportMapper.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
